harvester/DBInterface.py

```diff
@@ -26,34 +26,26 @@ def __init__(self, params):
-
-			cur.execute("CREATE TABLE IF NOT EXISTS repositories (repository_url TEXT, repository_name TEXT, repository_thumbnail TEXT, repository_type TEXT, last_crawl_timestamp NUMERIC, item_url_pattern TEXT, PRIMARY KEY (repository_url, repository_name))")
-
+			cur.execute("create table if not exists creators (creator_id INTEGER PRIMARY KEY NOT NULL,record_id INTEGER NOT NULL, creator TEXT, is_contributor INTEGER)")
+			cur.execute("create table if not exists descriptions (description_id INTEGER PRIMARY KEY NOT NULL,record_id INTEGER NOT NULL, description TEXT, language TEXT)")
+			cur.execute("create table if not exists domain_metadata (subject_id INTEGER PRIMARY KEY NOT NULL,record_id INTEGER NOT NULL, field_name TEXT, field_value TEXT)")
+			cur.execute("create table if not exists geospatial (subject_id INTEGER PRIMARY KEY NOT NULL,record_id INTEGER NOT NULL, coordinate_type TEXT, lat NUMERIC, lon NUMERIC)")
+			cur.execute("""create table if not exists records (record_id INTEGER PRIMARY KEY NOT NULL,repository_id INTEGER NOT NULL,title TEXT,pub_date TEXT,modified_timestamp INTEGER DEFAULT 0,
+				source_url TEXT,deleted NUMERIC DEFAULT 0,local_identifier TEXT,series TEXT,contact TEXT)""")
+			cur.execute("""create table if not exists repositories (repository_id INTEGER PRIMARY KEY NOT NULL,repository_set TEXT NOT NULL DEFAULT '',repository_url TEXT,repository_name TEXT,
+				repository_thumbnail TEXT,repository_type TEXT,last_crawl_timestamp INTEGER,item_url_pattern TEXT,abort_after_numerrors INTEGER,max_records_updated_per_run INTEGER,
+				update_log_after_numitems INTEGER,record_refresh_days INTEGER,repo_refresh_days INTEGER,enabled TEXT)""")
+			cur.execute("create table if not exists rights (subject_id INTEGER PRIMARY KEY NOT NULL,record_id INTEGER NOT NULL, rights TEXT)")
+			cur.execute("create table if not exists subjects (subject_id INTEGER PRIMARY KEY NOT NULL,record_id INTEGER NOT NULL, subject TEXT)")
+			cur.execute("create table if not exists tags (tag_id INTEGER PRIMARY KEY NOT NULL,record_id INTEGER NOT NULL, tag TEXT, language TEXT)")
+			cur.execute("create index if not exists creators_by_record on creators(record_id)")
+			cur.execute("create index if not exists descriptions_by_record on descriptions(record_id,language)")
+			cur.execute("create index if not exists tags_by_record on tags(record_id,language)")
+			cur.execute("create index if not exists subjects_by_record on subjects(record_id)")
+			cur.execute("create index if not exists rights_by_record on rights(record_id)")
+			cur.execute("create index if not exists geospatial_by_record on geospatial(record_id)")
+			cur.execute("create index if not exists domain_metadata_by_record on domain_metadata(record_id)")
+			cur.execute("create unique index if not exists records_by_repository on records (repository_id, local_identifier)")
 
```

Comment:
This is a lot harder to read than before, but not a huge problem; I don't think the original code was anywhere close to PEP8 so I don't have much of a defense.